File: ios/WebViewFramework/WebViewFramework/WebBridgeCallback.swift
//
//  WebBridgeCallback.swift
//  WebViewFramework
//
//  Created by Willy on 2025/5/26.
//

protocol WebBridgeCallback: AnyObject {
    func closeContainer()
    func openDeepLink(url: String)
    func logEvent(name: String, data: [String: Any]?)
    func refreshWebContainerConfig() -> WebContainerConfig
    func refreshToken() -> String
}

File: android/WebViewFramework/app/src/main/java/com/gogolook/webviewframework/WebBridgeCallback.kt
package com.gogolook.webviewframework

import org.json.JSONObject

interface WebBridgeCallback {
    fun closeContainer()
    fun openDeepLink(url: String)
    fun logEvent(name: String, data: JSONObject?)//TODO: Willy, JSONObject 改為 map??? 可以討論有沒有需要
    fun refreshWebContainerConfig(): WebContainerConfig
    fun refreshToken(): String
}